rofs/rofs.c

```diff
@@ -439,7 +439,7 @@ int rofs_lookup(struct rofs_ctx *ctx, oid_t *dir, const char *name, oid_t *fil,
 	int res;
 
 	fil->port = ctx->oid.port;
-	if (name == NULL) {
+	if ((name == NULL) || (name[0] == '\0')) {
```

Comment:
In case of `name[0] == '\0'`, `dirfind()` returns `-ENOENT`, so this should also be the case here. Maybe initialize `res` with `-ENOENT` or add a separate `if`?